text/0000-bundle-add.md

```diff
@@ -0,0 +1,132 @@
+- Start Date: 2016-12-04
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+Add a new `bundle add GEM` feature. The `bundle add GEM` command will add the gem to the gemfile (if valid) and run bundle install in one step. (Refer: [#4901](https://github.com/bundler/bundler/issues/4901)).
+
+The purpose of this RFC is to get feedback on use cases, cli format/syntax, and feature options.
+
+# Motivation
+
+The main use case is to deploy scripts and/or improve user productivity by adding a gem to Gemfile and bundle install in one step.
+
+The question is how far to extend this feature, is there a need to support all gem options like version, source, and group. And whether or not it should support multiple gems with their respective options.
+
+This command overlaps in function with the current `bundle inject GEM` command.
+
+# Detailed design
+
+The usage of the feature can be grouped into 4 categories:
+
+## 1. add a single gem
+#### Proposed command
+```ruby
+bundle add "GEM"
+
+#examples
+bundle add "rails"
+```
+#### Description
+This is the base use case. The following steps will occur when the command is executed:
+1. It will load the current Gemfile and resolve the new GEM with it.
+2. If it fails to resolve correctly or is an invalid GEM, it will error out.
+3. If it resolves successfully, it will install the latest version of the GEM.
+4. It will add the GEM to the Gemfile with a comment and place a conservative version update requirement e.g.
+```ruby
+  # Gemfile
+  # Added at 2016-12-07 00:21:12 -0800 by username:
+  gem "GEM", "~> 1.0.0"
```

Comment:
the version that is chosen depends on the version of the gem that is available. for example, if the version chosen by the resolver is < 1.0, the constraint should have three places e.g. version `0.12.3` becomes `~> 0.12.3`. If the resolved version is > 1.0, the constraint should only have two places e.g. version `1.2.3` becomes `~> 1.2`.